scheduler.go

```diff
@@ -576,8 +574,6 @@ func (s *Scheduler) TaskPresent(j interface{}) bool {
 }
 
 func (s *Scheduler) jobPresent(j *Job) bool {
-	s.jobsMutex.RLock()
```

Comment:
Actually looks like removing this lock is breaking the Update() func. Since it’s a private function we should be able to control any reads to it and if the race you’re seeing is related we can accomplish a different way

Reply:
@JohnRoesler, Yes jobPresent() is a private function and the read access of it is already controlled by s.Job() function.
An another way to solve this problem is, have another mutex which can be utilized but I don't see it is required though.

I am unable to locate the Update() func usage about this lock, and it is already using the s.Job() function to get a snapshot of the current running jobs. Please correct me if I am wrong, and happy to submit a new patch for this problem.

Reply:
There is a trace in the test results when running with test -race. You can see it in the actions output. I’m traveling and on mobile so I can’t dog on at the moment, but if you look there you’ll see why it’s failing the test with your change